libs/muddle/src/punishment_broadcast_channel.cpp

```diff
@@ -0,0 +1,252 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "muddle/punishment_broadcast_channel.hpp"
+
+using fetch::muddle::PunishmentBroadcastChannel;
+using fetch::muddle::QuestionStruct;
+
+QuestionStruct PunishmentBroadcastChannel::AllowPeerPull()
+{
+  FETCH_LOCK(lock_);
+  return question_;
+}
+
+PunishmentBroadcastChannel::PunishmentBroadcastChannel(Endpoint &endpoint, MuddleAddress address,
+                                                       CallbackFunction call_back,
+                                                       CertificatePtr certificate, uint16_t channel,
+                                                       bool ordered_delivery)
+  : endpoint_{endpoint}
+  , address_{std::move(address)}
+  , deliver_msg_callback_{std::move(call_back)}
+  , channel_{channel}
+  , certificate_{std::move(certificate)}
+  , rpc_client_{"PunishmentBC", endpoint_, SERVICE_PBC, channel_}
+  , state_machine_{std::make_shared<StateMachine>("PBCStateMach", State::INIT)}
+{
+  FETCH_UNUSED(ordered_delivery);
+  this->Expose(PULL_INFO_FROM_PEER, this, &PunishmentBroadcastChannel::AllowPeerPull);
```

Comment:
Is `this->` needed?

Reply:
No - I thought it improved readability since calling parent but taken it out